### ISSUE TITLE
AppState: document is not defined [server]

### ISSUE THREAD

**Issue by joncursi** [necolas/react-native-web#578]
I'm getting this error logged to my server console in version `0.0.116`. I am running a Meteor server which is crawls the code before delivering it to the browser. I am not using AppState directly at all, but since I am importing other components from `react-native-web` Meteor will try to crawl the files. Document is obviously not a global available on the server, so this breaks. Will likely also break for others trying to do SSR.

What can we do to fix this?

**Comment by joncursi** [comment]
Stack trace:

```
W20170727-18:00:39.420(-4)? (STDERR) /Users/jcursi/.meteor/packages/meteor-tool/.1.6.0-beta.4.mb3ak3++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle/server-lib/node_modules/fibers/future.js:280
W20170727-18:00:39.431(-4)? (STDERR) 						throw(ex);
W20170727-18:00:39.432(-4)? (STDERR) 						^
W20170727-18:00:39.433(-4)? (STDERR) 
W20170727-18:00:39.433(-4)? (STDERR) ReferenceError: document is not defined
W20170727-18:00:39.433(-4)? (STDERR)     at Object.<anonymous> (/Users/jcursi/Sites/joncursi/redbird-web/node_modules/react-native-web/dist/apis/AppState/index.js:1:1157)
W20170727-18:00:39.434(-4)? (STDERR)     at Module._compile (module.js:569:30)
W20170727-18:00:39.434(-4)? (STDERR)     at Object.Module._extensions..js (module.js:580:10)
W20170727-18:00:39.434(-4)? (STDERR)     at Module.load (module.js:503:32)
W20170727-18:00:39.435(-4)? (STDERR)     at tryModuleLoad (module.js:466:12)
W20170727-18:00:39.435(-4)? (STDERR)     at Function.Module._load (module.js:458:3)
W20170727-18:00:39.436(-4)? (STDERR)     at Module.require (module.js:513:17)
W20170727-18:00:39.436(-4)? (STDERR)     at require (internal/module.js:11:18)
W20170727-18:00:39.436(-4)? (STDERR)     at Object.<anonymous> (/Users/jcursi/Sites/joncursi/redbird-web/node_modules/react-native-web/dist/module.js:1:1487)
W20170727-18:00:39.436(-4)? (STDERR)     at Module._compile (module.js:569:30)
```

**Comment by joncursi** [comment]
I think this is the problem?

https://github.com/necolas/react-native-web/commit/0ff3e915926c35cad0a188b9ec3ffa9cbb62299a#diff-3b2bb0e3d72985cdc09ca57291fa6271R18

**Comment by joncursi** [comment]
Yeah, downgrading to `0.0.113`, which is just before that commit, resolves this issue. I think `document` is being called right there without first figuring out if the execution environment can understand the DOM.

**Comment by joncursi** [comment]
Solved in `0.0.117`, thank you!
